bootstrap/bootstrap.go

```diff
@@ -0,0 +1,321 @@
+}
+
+// CIPaths returns a Paths for a CI Job
+func CIPaths(base, job, build string) *Paths {
+	return &Paths{
+		Artifacts:   filepath.Join(base, job, build, "artifacts"),
+		BuildLog:    filepath.Join(base, job, build, "build-log.txt"),
+		Finished:    filepath.Join(base, job, build, "finished.json"),
+		Latest:      filepath.Join(base, job, "latest-build.txt"),
+		ResultCache: filepath.Join(base, job, "jobResultsCache.json"),
+		Started:     filepath.Join(base, job, build, "started.json"),
+	}
+}
+
+// PRPaths returns a Paths for a Pull Request
+func PRPaths(base string, repos Repos, job, build string) (*Paths, error) {
+	if len(repos) == 0 {
+		return nil, fmt.Errorf("repos should not be empty")
+	}
+	repo := repos.Main()
+	var prefix string
+	if repo.Name == "k8s.io/kubernetes" || repo.Name == "kubernetes/kubernetes" {
+		prefix = ""
+	} else if strings.HasPrefix(repo.Name, "k8s.io/") {
+		prefix = repo.Name[len("k8s.io/"):]
+	} else if strings.HasPrefix(repo.Name, "kubernetes/") {
+		prefix = repo.Name[len("kubernetes/"):]
+	} else if strings.HasPrefix(repo.Name, "github.com/") {
+		prefix = strings.Replace(repo.Name[len("github.com/"):], "/", "_", -1)
+	}
+	// Batch merges are those with more than one PR specified.
+	prNums := PullNumbers(repo.Pull)
+	var pull string
+	if len(prNums) > 1 {
+		pull = filepath.Join(prefix, "batch")
+	} else {
+		pull = filepath.Join(prefix, prNums[0])
+	}
+	prPath := filepath.Join(base, "pull", pull, job, build)
+	return &Paths{
+		Artifacts:     filepath.Join(prPath, "artifacts"),
+		BuildLog:      filepath.Join(prPath, "build-log.txt"),
+		PRPath:        prPath,
+		Finished:      filepath.Join(prPath, "finished.json"),
+		Latest:        filepath.Join(base, "directory", job, "latest-build.txt"),
+		PRBuildLink:   filepath.Join(base, "directory", job, build+".txt"),
+		PRLatest:      filepath.Join(base, "pull", pull, job, "latest-build.txt"),
+		PRResultCache: filepath.Join(base, "pull", pull, job, "jobResultsCache.json"),
+		ResultCache:   filepath.Join(base, "directory", job, "jobResultsCache.json"),
+		Started:       filepath.Join(prPath, "started.json"),
+	}, nil
+}
+
+// Bootstrap is the "real main" for bootstrap, after argument parsing
+func Bootstrap(args *Args) error {
+	repos, err := ParseRepos(args.Repo)
+	if err != nil {
+		return err
+	}
+
+	originalCWD, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get CWD! %v", err)
+	}
+
+	build := "fake" // TODO: port computing this value
+
+	var paths *Paths
+	if args.Upload != "" {
+		if repos.Main().Pull != "" {
+			paths, err = PRPaths(args.Upload, repos, args.Job, build)
+			if err != nil {
+				return err
+			}
+		} else {
+			paths = CIPaths(args.Upload, args.Job, build)
+		}
+		// TODO: Replace env var below with a flag eventually.
+		os.Setenv(GCSArtifactsEnv, paths.Artifacts)
+	}
+
+	// TODO(bentheelder): mimic the rest of bootstrap.py here ¯\_(ツ)_/¯
+	// Printing these so that it compiles ¯\_(ツ)_/¯
+	fmt.Println(originalCWD)
```

Comment:
unify with `log.Print()`?

Reply:
this is *only* so that it passes go tooling tests until these variables are actually in use.